chef/automate
# Store the organisation migration status in table

### :nut_and_bolt: Description: What code changed, and why?
Store the status of organisation migration from chef server to automate.



### :chains: Related Resources
https://github.com/chef/automate/issues/6377

### :+1: Definition of Done
I have added changes in get organisation API to store the status of organisation migration from chef server to automate.
and to do the migration asynchronously.
Also there should be single migration run at a time.
For load testing, added a command to  create the organisations on chef server.
For script, added an API to create the organisations on chef server.
 
### :athletic_shoe: How to Build and Test the Change
1. Get organisation API

    Steps to build:
    ```
      rebuild components/infra-proxy-service
      rebuild components/automate-gateway
    ```
    Steps to test:

    1. Ensure Chef Infra Server is running. If it is not start it with ```start_chef_server```
    2. Add the chef-server test environment credentials and extract the SERVER IDs

    Request
    This is a get request.
    ```curl -sSkH "api-token: $(get_admin_token)" 'https://a2-dev.test/api/v0/infra/servers/SERVER_ID/infraserverorgs'```

    Response

    ```
    {
        "migration_id": "9c5b892c-afbb-4df8-8da3-99919ae0cefc"
    }
    ```
2. Command to load the organisations

     Steps to build and test the changes:

     You need to enter the hab studio then check the services are running.
     To start the services you can use ```start_infra_proxy_service``` or ```start_all_services```
     Then run ```infra_service_load_sample_orgs```

    Options:
    -N: Number of records, default 50
    -S: Chef infra server id, which should be available on chef server and added in automate
    -O: Chef organization's name prefix, default: chef-org

    Example:
    ```infra_service_load_sample_orgs -N 50 -S local-dev -O chef-org ```

### :white_check_mark: Checklist

**All PRs** must tick these:


With occasional exceptions, all PRs **from Progress employees** must tick these:


All PRs **from Progress employees** should tick these if appropriate:


*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
